src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -266,6 +267,14 @@ def __init__(
         # UUID for the plan to uniquely identify the SnowflakePlan object. We also use this
         # to UUID track queries that are generated from the same plan.
         self._uuid = str(uuid.uuid4())
+        self._attributes = None
+        self._quoted_identifiers = None
+        # If _attributes is not None, then _quoted_identifiers will be None.
+        # If _quoted_identifiers is not None, then _attributes will be None.
+        if session.reduce_describe_query_enabled and self.source_plan is not None:
+            self._attributes, self._quoted_identifiers = infer_metadata(
+                self.source_plan
+            )
 
```

Comment:
let's also do a post check  that 
1) if the flag is off, none of those fields can be not None
2) the _attribute and _quited_identifier will not both be not None

Reply:
sure